lib/vanagon/driver.rb

```diff
@@ -76,14 +76,17 @@ def run
       @engine.startup(@workdir)
 
       puts "Target is #{@engine.target}"
-
-      install_build_dependencies
+      Vanagon::Utilities.retry_with_timeout(3, 3600) do
+        install_build_dependencies
+      end
       @project.fetch_sources(@workdir)
       @project.make_makefile(@workdir)
       @project.make_bill_of_materials(@workdir)
       @project.generate_packaging_artifacts(@workdir)
       @engine.ship_workdir(@workdir)
-      @engine.dispatch("(cd #{@engine.remote_workdir}; #{@platform.make})")
+      Vanagon::Utilities.retry_with_timeout(3, 3600) do
+        @engine.dispatch("(cd #{@engine.remote_workdir}; #{@platform.make})")
+      end
```

Comment:
I was actually thinking something similar but had no idea how to mock it up, I am very +1 to something like this